src/App.js

```diff
@@ -1,13 +1,162 @@
 import React, { Component } from 'react';
+import ContactList from './ContactList';
+import SearchBar from './SearchBar';
+import ContactForm from './ContactForm';
+import axios from 'axios';
 
 class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      searchText: '',
+      contacts: [],
+      addedContacts: [],
+    };
+  }
+
+  handleSearchBarChange(event) {
+    this.setState({
+      searchText: event.target.value
+    });
+  }
+
+  componentDidMount() {
+    axios.get('http://localhost:4000/contacts')
+      .then(resp => {
+        this.setState({
+          contacts: resp.data
+        });
+      })
+      .catch(err => {
+        console.log(`Error! ${err}`)
+      });
+  }
+
+  handleChange(event) {
+    this.setState({
+      searchText: event.target.value
+    });
+  }
+
+  componentWillMount() {
+    console.log('componentWillMount');
+  }
+
+  getFilteredContacts() {
+
+    const term = this.state.searchText.trim().toLowerCase();
+    const contacts = this.state.contacts;
+
+    if (!term) {
+      return contacts;
+    }
+
+    return contacts.filter(contact => {
+      return contact.name.toLowerCase().indexOf(term) >= 0;
+    });
+  }
+
+  handleAddContact(attributes) {
+    axios.post('http://localhost:4000/contacts', attributes)
+      .then(resp => {
+        this.setState({
+          contacts: [...this.state.contacts, resp.data]
+        });
+      })
+      .catch(err => console.log(err));
+  }
+
+  addContact(contact) {
+    const curContacts = [...this.state.contacts];
+    const curContact = curContacts[contact - 1];
+    const added = [...this.state.addedContacts];
+    const newContact = {
+      _id: curContact._id + 10,
+      name: curContact.name,
+      occupation: curContact.occupation,
+      avatar: curContact.avatar,
+      active: '',
+      buttonType: 'Remove'
```

Comment:
When using types like this, I wouldn't put them in hardcoded strings. Better define them as a const outside of the class. That way this is easier to refactor: 

```javascript
const BUTTON_TYPE_REMOVE = 1; 
```
This would also be better placed inside of the button component.